api/types.proto

```diff
@@ -506,6 +506,21 @@ message PortConfig {
 		UDP = 1 [(gogoproto.enumvalue_customname) = "ProtocolUDP"];
 	}
 
+	enum Exposition {
+		option (gogoproto.enum_customname) = "PortConfigExposition";
+		option (gogoproto.goproto_enum_prefix) = false;
+
+		// PortConfigExpositionIngress exposes the port across the cluster on
+		// all nodes.
+		INGRESS = 0;
+
+		// PortConfigExpositionHost exposes the port on just the target host.
+		// If the published port is undefined, an ephemeral port will be
+		// allocated. If the published port is defined, the node will attempt
```

Comment:
Yes, this is the part we kind of discussed in the meeting and it was felt that the major use case is to have ephemeral port mapping capability as long as it is query-able at the task level from docker api. Later on not sure if these can be tied to a scheduler constraint since the manager would effectively know about all the ports that swarm tasks bind to on any node if some service wants to use a specific port.